integration/query_frontend_test.go

```diff
@@ -560,40 +560,74 @@ overrides:
+	runQueryFrontendWithQueryShardingHTTPTest(
+		t,
+		queryFrontendTestConfig{
+			querySchedulerEnabled: false,
 			setup: func(t *testing.T, s *e2e.Scenario) (configFile string, flags map[string]string) {
 				flags = mergeFlags(BlocksStorageFlags(), BlocksStorageS3Flags(), map[string]string{
-					"-query-scheduler.max-outstanding-requests-per-tenant": "1",
+					// Set the maximum entry size to 50 byte.
+					// The query size is 64 bytes, so it will be a too large entry request.
+					"-querier.frontend-client.grpc-max-send-msg-size": "50",
 				})
 
 				minio := e2edb.NewMinio(9000, flags["-blocks-storage.s3.bucket-name"])
 				require.NoError(t, s.StartAndWaitReady(minio))
 
 				return "", flags
 			},
-		})
+		},
+		http.StatusRequestEntityTooLarge,
+		false,
+	)
+}
+
+func TestQueryFrontendWithQueryShardingAndTooManyRequests(t *testing.T) {
+	t.Run("with query-scheduler", func(t *testing.T) {
+		runQueryFrontendWithQueryShardingHTTPTest(
+			t,
+			queryFrontendTestConfig{
+				querySchedulerEnabled: true,
+				setup: func(t *testing.T, s *e2e.Scenario) (configFile string, flags map[string]string) {
+					flags = mergeFlags(BlocksStorageFlags(), BlocksStorageS3Flags(), map[string]string{
+						"-query-scheduler.max-outstanding-requests-per-tenant": "1",
+					})
+
+					minio := e2edb.NewMinio(9000, flags["-blocks-storage.s3.bucket-name"])
+					require.NoError(t, s.StartAndWaitReady(minio))
+
+					return "", flags
+				},
+			},
+			http.StatusTooManyRequests,
+			true,
+		)
 	})
 
 	t.Run("without query-scheduler", func(t *testing.T) {
-		runQueryFrontendWithQueryShardingAndTooManyRequestsTest(t, queryFrontendTestConfig{
-			querySchedulerEnabled: false,
-			setup: func(t *testing.T, s *e2e.Scenario) (configFile string, flags map[string]string) {
-				flags = mergeFlags(BlocksStorageFlags(), BlocksStorageS3Flags(), map[string]string{
-					"-querier.max-outstanding-requests-per-tenant": "1", // Limit
-				})
-
-				minio := e2edb.NewMinio(9000, flags["-blocks-storage.s3.bucket-name"])
-				require.NoError(t, s.StartAndWaitReady(minio))
-
-				return "", flags
+		runQueryFrontendWithQueryShardingHTTPTest(
+			t,
+			queryFrontendTestConfig{
+				querySchedulerEnabled: false,
+				setup: func(t *testing.T, s *e2e.Scenario) (configFile string, flags map[string]string) {
+					flags = mergeFlags(BlocksStorageFlags(), BlocksStorageS3Flags(), map[string]string{
+						"-querier.max-outstanding-requests-per-tenant": "1", // Limit
+					})
+
+					minio := e2edb.NewMinio(9000, flags["-blocks-storage.s3.bucket-name"])
+					require.NoError(t, s.StartAndWaitReady(minio))
+
+					return "", flags
+				},
 			},
-		})
+			http.StatusTooManyRequests,
+			true,
+		)
 	})
 }
-func runQueryFrontendWithQueryShardingAndTooManyRequestsTest(t *testing.T, cfg queryFrontendTestConfig) {
+func runQueryFrontendWithQueryShardingHTTPTest(t *testing.T, cfg queryFrontendTestConfig,
+	exceptHTTPSStatus int, checkDiscardedMetrics bool) {
```

Comment:
* a new line on top
* I don't think we use the line breaking that much in this repo
* there's a typo in exceptHTTPSStatus -> expectHTTPSStatus

```suggestion

func runQueryFrontendWithQueryShardingHTTPTest(t *testing.T, cfg queryFrontendTestConfig, expectHTTPSStatus int, checkDiscardedMetrics bool) {
```

Reply:
Fixed. Thanks!